### PR TITLE
Separate models in indirect data analysis - IndirectFitDataPresenter

### PR DESCRIPTION
**Pull request by Harrietbrown**
**Description of work.**

This PR separates aspects of the IndirectFitDataPresenter from the IndirectFittingModel and merges it with IndirectFitDataTable in to it's own model.

**To test:**

There should be no changes to fucntionality with this PR. It is important to check that the standard usage of all the tabs opperate correctly.

Partial #29210

*This does not require release notes* because **It is entirly internal changes**

Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/Indirect/FqFitDataView.cpp
Comment (sf1919) [inline qt/scientific_interfaces/Indirect/FqFitDataView.cpp:3]: Should the date be 2021?

File: qt/scientific_interfaces/Indirect/FqFitModel.cpp
Comment (sf1919) [inline qt/scientific_interfaces/Indirect/FqFitModel.cpp:355]: Most of FqFitModel has been removed. Is there a reason to still have a separate FqFitModel?

**Review by sf1919 — CHANGES_REQUESTED** [review]
I have manually tested all the tabs in the Indirect Data Analysis GUI and they all work as expected. Even the new Elwin tab functions as expected with recent changes made in PR #32044 .

Reviewing the code I have a couple of comments though which shouldn't take long to resolve.

File: qt/scientific_interfaces/Indirect/FqFitModel.cpp
Comment (Harrietbrown) [reply qt/scientific_interfaces/Indirect/FqFitModel.cpp:355]: The three functions remaining are required to be defined seperatly. so we do still need FqFitModel

File: qt/scientific_interfaces/Indirect/FqFitDataView.cpp
Comment (Harrietbrown) [reply qt/scientific_interfaces/Indirect/FqFitDataView.cpp:3]: I can change that

File: qt/scientific_interfaces/Indirect/FqFitModel.cpp
Comment (sf1919) [reply qt/scientific_interfaces/Indirect/FqFitModel.cpp:355]: Fair enough! Thought I would double check to be on the safe side.

**Review by sf1919 — APPROVED** [review]
As already commenetd above the changes do not break the functionality of the Indirect Data Analysis GUIs - all work as intended. The changes made do streamline the Indirect code base and will make it much easier for developers to work with in future. A much needed change that has been well done.

**Comment by DavidFair** [comment]
I haven't reviewed this yet

@Harrietbrown 
Since this is effectively maintenance (since there are no user visible changes) can we defer it until after the release-next branch has been created? 
That way if there are any regressions we can find them in the following 3 months, as this is a very large diff. As soon as release-next is created I'm happy to get this through so you have a good starting point for the next sprint

**Comment by Harrietbrown** [comment]
> I haven't reviewed this yet
> 
> @Harrietbrown
> Since this is effectively maintenance (since there are no user visible changes) can we defer it until after the release-next branch has been created?
> That way if there are any regressions we can find them in the following 3 months, as this is a very large diff. As soon as release-next is created I'm happy to get this through so you have a good starting point for the next sprint

There is going to be one more PR as part of the refactor that I will be working on during maininance. It would be best to get this in mow so it will not hold up the next PR.

**Comment by martyngigg** [comment]
@Harrietbrown Can you start you next branch based off this one to continue working? The `release-next` branch will be created on Monday anyway so we can merge this to `master` as soon as it's created.

File: qt/scientific_interfaces/Indirect/test/ConvFitDataPresenterTest.h
Comment (DavidFair) [inline qt/scientific_interfaces/Indirect/test/ConvFitDataPresenterTest.h:49]: Just a heads up, but not something you need to do, this is the old style Gmock decelerations. The newer ones are much easier to work with as you don't need to count params or check if it's a const type

```c++
// Interface.h
virtual void foo(int, double &) const = 0;

// Test file
MOCK_METHOD(void, foo, (int, double&), (const, override));
```

**Review by DavidFair — APPROVED** [review]
LGTM, there is some larger blocks in the diff that have been moved around. Just for reference, anything that was moved "as-is" I've avoided reviewing since it's effectively unchanged from before

**Comment by martyngigg** [comment]
Are we agreed that we will merge this to `master` **after** `release-next` is created? If so I'll move the milestone to 6.3.

**Comment by Harrietbrown** [comment]
Yes we will merge after release next is made.

**Comment by DavidFair** [comment]
`release-next` has been made so I'll merge this now before I forget.
Sorry if this caused any problems, the joys of releases :sweat_smile:

**Pull request merged**
